web/packages/teleterm/src/ui/Vnet/DocumentVnetDiagReport.tsx

```diff
@@ -0,0 +1,301 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import styled from 'styled-components';
+
+import { Button, Alert as DesignAlert, Flex, H1, Link, Stack } from 'design';
+import { AlertProps } from 'design/Alert/Alert';
+import Table, { TextCell } from 'design/DataTable';
+import { displayDateTime } from 'design/datetime';
+import {
+  Copy,
+  Download,
+  Refresh,
+  Check as SuccessIcon,
+  Warning as WarningIcon,
+} from 'design/Icon';
+import { P1, P2 } from 'design/Text/Text';
+import { HoverTooltip } from 'design/Tooltip';
+import { Timestamp } from 'gen-proto-ts/google/protobuf/timestamp_pb';
+import * as diag from 'gen-proto-ts/teleport/lib/vnet/diag/v1/diag_pb';
+import { pluralize } from 'shared/utils/text';
+
+import { reportOneOfIsRouteConflictReport } from 'teleterm/helpers';
+import Document from 'teleterm/ui/Document';
+import type * as docTypes from 'teleterm/ui/services/workspacesService';
+
+export function DocumentVnetDiagReport(props: {
+  visible: boolean;
+  doc: docTypes.DocumentVnetDiagReport;
+}) {
+  const { report } = props.doc;
+  const { networkStackAttempt } = report;
+  const { networkStack } = networkStackAttempt;
+  const createdAt = displayDateTime(Timestamp.toDate(report.createdAt));
+
+  return (
+    <Document visible={props.visible}>
+      <Stack
+        gap={4}
+        maxWidth="680px"
+        width="100%"
+        mx="auto"
+        mt={4}
+        p={5}
+        backgroundColor="levels.surface"
+        borderRadius={2}
+        // Without this, the Stack would span the whole height of the Document, no matter how much
+        // content was displayed in the Stack.
+        alignSelf="flex-start"
+      >
+        <Stack gap={2} width="100%" alignItems="stretch">
+          <Flex
+            flexWrap="wrap"
+            width="100%"
+            gap={2}
+            justifyContent="space-between"
+          >
+            <H1>VNet Diagnostic Report</H1>
+
+            <Flex gap={2}>
+              {/* TODO: Implement buttons. */}
+              <HoverTooltip tipContent="Run Diagnostics Again">
+                <Button intent="neutral" p={1}>
+                  <Refresh size="medium" />
+                </Button>
+              </HoverTooltip>
+
+              <HoverTooltip tipContent="Copy Report to Clipboard">
+                <Button intent="neutral" p={1}>
+                  <Copy size="medium" />
+                </Button>
+              </HoverTooltip>
+
+              <HoverTooltip tipContent="Save Report to File">
+                <Button intent="neutral" p={1}>
+                  <Download size="medium" />
+                </Button>
+              </HoverTooltip>
+            </Flex>
+          </Flex>
+
+          {networkStackAttempt.status === diag.CheckAttemptStatus.ERROR && (
+            <>
+              <P2>Created at: {createdAt}</P2>
+              <Alert
+                kind="danger"
+                details={<P2>{networkStackAttempt.error}</P2>}
```

Comment:
Do we need this (and others) `P2`? I always put the text directly into `details` :p
 From what I see, it looks the same without it.

Reply:
I'd rather put text within `<p>` rather than `<div>`. 😏